src/server/worker/pipeline/common.go

```diff
@@ -54,6 +122,95 @@ func RunUserCode(
 	})
 }
 
+func processFiles(ctx context.Context,
+	pachClient *client.APIClient,
+	pipelineInfo *pps.PipelineInfo,
+	logger logs.TaggedLogger) (retErr1 error) {
+	repo := pipelineInfo.Pipeline.Name
+	var commit *pfs.Commit
+	var err error
+	for _, f := range fileList {
+		if commit == nil {
```

Comment:
I think this loop creates one commit for all the files received in on batch. Is that correct? IIRC, we want to create a commit for each tar file we receive.

Reply:
yeah, this looks right